docs/tls-authentication.md

```diff
@@ -0,0 +1,248 @@
+# Authentication
+
+To defend against impersonation attacks, as well as to maintain an encrypted communication
+between the wallet's frontend (a.k.a. _[daedalus](https://github.com/input-output-hk/daedalus)_)
+and the wallet's backend (a.k.a _[cardano-sl](https://github.com/input-output-hk/cardano-sl)_) both parties rely on TLS certificates and mututal
+verification of their identity. 
+
+
+## Table of Contents
+
+  * [Motivation and Design](#motivation-and-design)
+
+  * [Prerequisites](#prerequisites)
+
+  * [Generating Certificates](#generating-certificates)
+    + [Default (localhost)](#quick-and-easy)
+    + [Personalized (custom domain)](#step-by-step)
+
+  * [Using Certificates](#using-certificates)
+    + [Setting Up Cardano-SL Backend](#setting-up-cardano-sl-backend)
+    + [Contacting Cardano-SL Backend](#contacting-cardano-sl-backend)
+
+
+## Motivation and Design
+
+Communication between _daedalus_ and _cardano-sl_ is done via http server. There are various
+attacks possible to this kind of environment. In order to secure the connection between these
+two services, we rely on TLS and X.509 public key certificates signed by a known authority. 
+
+Hence, the backend server expects clients to provide valid X.509 v3 certificates signed by an
+approved authority (identified by another certificate, provided on initialization) to verify
+their identity. On the other hand, clients must check the certificate provided by the server to
+make sure they're dealing with the expected party and not a malicious service trying to hijack
+the backend's identity. 
+
+Certificates (and corresponding private keys) are generated off-band, and signed by a same CA's
+(e.g. _Certificate Authority_) private key. The steps below describe how to securely generate
+them. Securing the access to the provided private keys is of vitale importance in order to keep
+the system safe. **Any process or machine with access to the private keys may be able to
+usurp daedalus or cardano-sl's identities**.
+
+
+## Prerequisites
+
+- Install and configure [OpenSSL](https://www.openssl.org/)
```

Comment:
We already have a requirement of nix to build `cardano-sl` for exchanges. Lets just ship any scripts needed with a shebang to include openssl as a dependency.

Reply:
Maybe have a look at [CBR-154](https://iohk.myjetbrains.com/youtrack/issue/CBR-154) ?